File: Domains/Base/Database/Server/Allors/Base/Database/Pull/PullResponse.cs
namespace Allors.Web.Database
{
    using System.Collections.Generic;

    public class PullResponse
    {
        public string UserSecurityHash { get; set; }

        public string[][] Objects { get; set; }

        public Dictionary<string, string> NamedObjects { get; set; }

        public Dictionary<string, string[]> NamedCollections { get; set; }

        public Dictionary<string, object> NamedValues { get; set; }
    }
}

File: Domains/Custom/Workspace/Typescript/Angular/src/domain/base/data/responses/PullResponse.ts
import { Response } from "./Response";

export interface PullResponse extends Response {
    userSecurityHash: string;
    objects?: string[][];

    namedObjects?: { [id: string]: string; };
    namedCollections?: { [id: string]: string[]; };
    namedValues?: { [id: string]: any; };
}
